### PR TITLE
fix(position): [#FOR-559] fix unique position between question and section

### PR DESCRIPTION
**Pull request by qperie**
## Describe your changes
Two problems were fiwed here :
- First, we could switch two sections, now we can
- Second, we added checks to avoid having a question and a section with the same position in the same form. We check it in Java, because we can't add a SQL contraint on columns of two different tables

There's no TU added here, you can find them in the next PR -> #215 

## Issue ticket number and link
FOR-559 : https://jira.support-ent.fr/browse/FOR-559

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by qpic-cgi — APPROVED** [review]

File: common/src/main/java/fr/openent/form/core/models/FormElement.java [hunk 1]
```diff
@@ -64,5 +70,36 @@ public JsonObject toJson() {
...
+        return formElements.stream()
+                .map(formElement -> {
+                    if (formElement instanceof JsonObject) {
+                        JsonObject jsonFormElement = (JsonObject)formElement;
+                        return jsonFormElement.containsKey(DESCRIPTION) ? new Section(jsonFormElement) : new Question(jsonFormElement);
+                    }
+                    return null;
+                })
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
```
[the remark this answers is not in the document]
Comment (qpic-cgi) [reply common/src/main/java/fr/openent/form/core/models/FormElement.java:84]: tu es obliger de chainer tes maps ?

map(formElement -> {
       JsonObject jsonFormElement = JsonObject.class.cast(formElement);
       jsonFormElement.containsKey(DESCRIPTION) ? new Section(jsonFormElement) : new Question(jsonFormElement));
})

ça va plus vite !

File: common/src/main/java/fr/openent/form/core/models/FormElement.java [hunk 2]
```diff
@@ -64,5 +70,36 @@ public JsonObject toJson() {
...
+    }
+
+    public static JsonArray toJsonArrayFormElements(List<FormElement> formElements) {
+        return new JsonArray(formElements.stream().map(FormElement::toJson).collect(Collectors.toList()));
+    }
+
+    public static List<Section> getSections(List<FormElement> formElements) {
+        return formElements.stream()
+                .filter(e -> e instanceof Section)
+                .map(e -> (Section)e)
```
[the remark this answers is not in the document]
Comment (qperie) [reply common/src/main/java/fr/openent/form/core/models/FormElement.java]: C'est pour renvoyer une List<Section> au lieu d'une List<FormElement>

File: common/src/main/java/fr/openent/form/core/models/FormElement.java [hunk 3]
```diff
@@ -64,5 +70,36 @@ public JsonObject toJson() {
                 .put(TITLE, this.title)
                 .put(POSITION, this.position);
     }
+
+    public static List<FormElement> toListFormElements(JsonArray formElements) {
+        return formElements.stream()
+                .map(formElement -> {
+                    if (formElement instanceof JsonObject) {
+                        JsonObject jsonFormElement = (JsonObject)formElement;
+                        return jsonFormElement.containsKey(DESCRIPTION) ? new Section(jsonFormElement) : new Question(jsonFormElement);
```
[the remark this answers is not in the document]
Comment (qperie) [reply common/src/main/java/fr/openent/form/core/models/FormElement.java]: J'aime bien l'idée du slug mais si ça fera un peu redondant dans la base. J'vais voir si je peux ajouter ça parce que j'suis d'accord c'est pas très beau comme ça 👍

File: common/src/main/java/fr/openent/form/core/models/FormElement.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (qperie) [reply common/src/main/java/fr/openent/form/core/models/FormElement.java]: Nope, il me met erreur sinon, il attend vraiment des types Section. Sinon pour lui un FormElement peut toujours être éventuellement un type Question je suppose

[the remark this answers is not in the document]
Comment (qpic-cgi) [reply common/src/main/java/fr/openent/form/core/models/FormElement.java]: okey okey, ça me surprend ! ça marche dans ce cas

**Pull request merged**